### PR TITLE
Delete deprecated starboard header files

### PR DESCRIPTION
**Pull request by madhurajayaraman**
b/362374622

### CODE REVIEW

File: base/files/file_enumerator.h [hunk 1]
```diff
@@ -20,7 +20,7 @@
 #if defined(STARBOARD)
 #include <sys/stat.h>
 #include <unistd.h>
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline base/files/file_enumerator.h:23]: Small nit, but you have a blank line left in a few places here where we don't need it. I'll just add "remove" below for those.

File: base/files/file_enumerator_starboard.cc [hunk 2]
```diff
@@ -22,8 +22,7 @@
 #include "starboard/common/log.h"
 #include "starboard/common/string.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/directory.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline base/files/file_enumerator_starboard.cc:25]: remove

File: base/files/file_starboard.cc [hunk 3]
```diff
@@ -28,7 +28,7 @@
 #include "base/strings/stringprintf.h"
 #include "base/threading/thread_restrictions.h"
 #include "starboard/common/metrics/stats_tracker.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline base/files/file_starboard.cc:31]: remove

File: base/files/platform_file.h [hunk 4]
```diff
@@ -9,7 +9,7 @@
 #include "build/build_config.h"
 
 #if defined(STARBOARD)
```
Comment (kaidokert) [inline base/files/platform_file.h:11]: These #ifdef(STARBOARD) are no longer needed

File: base/files/scoped_file.h [hunk 5]
```diff
@@ -15,7 +15,7 @@
 #include "build/build_config.h"
 
 #if defined(STARBOARD)
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline base/files/scoped_file.h:18]: remove

File: base/logging.cc [hunk 6]
```diff
@@ -48,7 +48,7 @@
 #include "starboard/common/time.h"
 #include "starboard/configuration.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline base/logging.cc:51]: remove

File: base/memory/platform_shared_memory_handle.h [hunk 7]
```diff
@@ -20,7 +20,7 @@
 #include "base/files/scoped_file.h"
 #elif defined(STARBOARD)
 #include "base/files/scoped_file.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline base/memory/platform_shared_memory_handle.h:23]: remove

File: chrome/updater/util.cc [hunk 8]
```diff
@@ -23,8 +23,7 @@
 #include "crypto/sha2.h"
 #include "starboard/configuration_constants.h"
 #include "starboard/extension/installation_manager.h"
-#include "starboard/file.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline chrome/updater/util.cc:26]: remove

File: cobalt/dom/window.cc [hunk 9]
```diff
@@ -57,7 +57,7 @@
 #include "cobalt/web/environment_settings_helper.h"
 #include "cobalt/web/event.h"
 #include "cobalt/web/window_or_worker_global_scope.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline cobalt/dom/window.cc:60]: remove

File: cobalt/loader/image/jpeg_image_decoder.h [hunk 10]
```diff
@@ -23,7 +23,7 @@
 #include "base/callback.h"
 #include "cobalt/loader/image/image.h"
 #include "cobalt/loader/image/image_data_decoder.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline cobalt/loader/image/jpeg_image_decoder.h:26]: remove

File: cobalt/network/dial/dial_system_config.cc [hunk 11]
```diff
@@ -25,7 +25,7 @@
 #include "net/base/net_string_util.h"
 
 #if defined(STARBOARD)
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline cobalt/network/dial/dial_system_config.cc:28]: remove

File: components/update_client/cobalt_slot_management.cc [hunk 12]
```diff
@@ -25,7 +25,7 @@
 #include "chrome/updater/util.h"
 #include "components/update_client/utils.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline components/update_client/cobalt_slot_management.cc:28]: remove

File: net/cert/internal/trust_store_in_memory_starboard.cc [hunk 13]
```diff
@@ -27,9 +27,8 @@
 #include "net/cert/x509_util.h"
 #include "starboard/common/string.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/directory.h"
-#include "starboard/file.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline net/cert/internal/trust_store_in_memory_starboard.cc:30]: remove

File: starboard/android/shared/android_main.cc [hunk 14]
```diff
@@ -27,8 +27,7 @@
 #include "starboard/common/semaphore.h"
 #include "starboard/common/string.h"
 #if SB_IS(EVERGREEN_COMPATIBLE)
-#include "starboard/directory.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/android/shared/android_main.cc:30]: remove

File: starboard/android/shared/video_decoder.cc [hunk 15]
```diff
@@ -36,7 +36,7 @@
 #include "starboard/memory.h"
 #include "starboard/shared/starboard/media/mime_type.h"
 #include "starboard/shared/starboard/player/filter/video_frame_internal.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/android/shared/video_decoder.cc:39]: remove

File: starboard/client_porting/icu_init/icu_init.cc [hunk 16]
```diff
@@ -23,7 +23,7 @@
 #include "starboard/common/once.h"
 #include "starboard/configuration.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/client_porting/icu_init/icu_init.cc:26]: remove

File: starboard/common/paths.cc [hunk 17]
```diff
@@ -20,7 +20,7 @@
 #include <vector>
 #include "starboard/common/log.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/common/paths.cc:23]: remove

File: starboard/contrib/cast/cast_starboard_api/samples/cast_starboard_api_test.cc [hunk 18]
```diff
@@ -8,7 +8,7 @@
 #include "starboard/common/thread.h"
 #include "starboard/egl.h"
 #include "starboard/event.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/contrib/cast/cast_starboard_api/samples/cast_starboard_api_test.cc:11]: remove

File: starboard/elf_loader/elf_loader_impl.cc [hunk 19]
```diff
@@ -25,7 +25,7 @@
 #include "starboard/elf_loader/lz4_file_impl.h"
 #include "starboard/extension/loader_app_metrics.h"
 #include "starboard/memory.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/elf_loader/elf_loader_impl.cc:28]: remove

File: starboard/elf_loader/elf_loader_sandbox.cc [hunk 20]
```diff
@@ -24,7 +24,7 @@
 #include "starboard/elf_loader/sabi_string.h"
 #include "starboard/event.h"
 #include "starboard/shared/starboard/command_line.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/elf_loader/elf_loader_sandbox.cc:27]: remove

File: starboard/elf_loader/exported_symbols.cc [hunk 21]
```diff
@@ -57,7 +54,7 @@
 #include "starboard/socket_waiter.h"
 #include "starboard/speech_synthesis.h"
 #include "starboard/storage.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/elf_loader/exported_symbols.cc:57]: remove

File: starboard/elf_loader/lz4_file_impl_test.cc [hunk 22]
```diff
@@ -17,7 +17,7 @@
 #include "starboard/elf_loader/lz4_file_impl.h"
 
 #include "starboard/configuration_constants.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/elf_loader/lz4_file_impl_test.cc:20]: remove

File: starboard/elf_loader/relocations_test.cc [hunk 23]
```diff
@@ -18,7 +18,7 @@
 
 #include "starboard/elf_loader/elf.h"
 #include "starboard/elf_loader/file_impl.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/elf_loader/relocations_test.cc:21]: remove

File: starboard/loader_app/loader_app.cc [hunk 24]
```diff
@@ -28,7 +28,7 @@
 #include "starboard/elf_loader/sabi_string.h"
 #include "starboard/event.h"
 #include "starboard/extension/loader_app_metrics.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/loader_app/loader_app.cc:31]: remove

File: starboard/loader_app/loader_app.cc [hunk 25]
```diff
@@ -38,7 +38,7 @@
 #include "starboard/loader_app/system_get_extension_shim.h"
 #include "starboard/memory.h"
 #include "starboard/shared/starboard/command_line.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/loader_app/loader_app.cc:41]: remove

File: starboard/loader_app/slot_management.cc [hunk 26]
```diff
@@ -31,12 +31,12 @@
 #include "starboard/elf_loader/sabi_string.h"
 #include "starboard/event.h"
 #include "starboard/extension/loader_app_metrics.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/loader_app/slot_management.cc:34]: remove

File: starboard/loader_app/slot_management_test.cc [hunk 27]
```diff
@@ -26,10 +26,9 @@
 #include "gmock/gmock.h"
 #include "starboard/common/file.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/directory.h"
 #include "starboard/elf_loader/sabi_string.h"
 #include "starboard/event.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/loader_app/slot_management_test.cc:31]: remove

File: starboard/loader_app/system_get_extension_shim.cc [hunk 28]
```diff
@@ -20,7 +20,7 @@
 #include "starboard/common/log.h"
 #include "starboard/extension/installation_manager.h"
 #include "starboard/loader_app/installation_manager.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/loader_app/system_get_extension_shim.cc:23]: remove

File: starboard/nplb/extern_c_test.cc [hunk 29]
```diff
@@ -17,30 +17,27 @@
...
 #include "starboard/configuration.h"
 #include "starboard/cpu_features.h"
 #include "starboard/decode_target.h"
-#include "starboard/directory.h"
 #include "starboard/drm.h"
 #include "starboard/egl.h"
 #include "starboard/event.h"
 #include "starboard/export.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/nplb/extern_c_test.cc:27]: remove

File: starboard/nplb/extern_c_test.cc [hunk 30]
```diff
@@ -17,30 +17,27 @@
...
 #include "starboard/memory.h"
 #include "starboard/microphone.h"
-#include "starboard/mutex.h"
 #include "starboard/player.h"
 #include "starboard/socket.h"
 #include "starboard/socket_waiter.h"
 #include "starboard/speech_synthesis.h"
 #include "starboard/storage.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/nplb/extern_c_test.cc:40]: remove

File: starboard/nplb/file_delete_recursive_test.cc [hunk 31]
```diff
@@ -18,8 +18,7 @@
 
 #include "starboard/common/file.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/directory.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/nplb/file_delete_recursive_test.cc:21]: remove

File: starboard/nplb/include_all.c [hunk 32]
```diff
@@ -16,30 +16,27 @@
...
 #include "starboard/configuration.h"
 #include "starboard/cpu_features.h"
 #include "starboard/decode_target.h"
-#include "starboard/directory.h"
 #include "starboard/drm.h"
 #include "starboard/egl.h"
 #include "starboard/event.h"
 #include "starboard/export.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/nplb/include_all.c:26]: remove

File: starboard/nplb/include_all.c [hunk 33]
```diff
@@ -16,30 +16,27 @@
...
 #include "starboard/memory.h"
 #include "starboard/microphone.h"
-#include "starboard/mutex.h"
 #include "starboard/player.h"
 #include "starboard/socket.h"
 #include "starboard/socket_waiter.h"
 #include "starboard/speech_synthesis.h"
 #include "starboard/storage.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/nplb/include_all.c:39]: remove

File: starboard/nplb/nplb_evergreen_compat_tests/fonts_test.cc [hunk 34]
```diff
@@ -19,7 +19,7 @@
 
 #include "starboard/common/log.h"
 #include "starboard/configuration.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/nplb/nplb_evergreen_compat_tests/fonts_test.cc:22]: remove

File: starboard/nplb/player_get_audio_configuration_test.cc [hunk 35]
```diff
@@ -15,7 +15,7 @@
 #include <vector>
 
 #include "starboard/nplb/player_test_fixture.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/nplb/player_get_audio_configuration_test.cc:18]: remove

File: starboard/nplb/player_get_media_time_test.cc [hunk 36]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 #include "starboard/nplb/player_test_fixture.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/nplb/player_get_media_time_test.cc:16]: remove

File: starboard/nplb/player_write_sample_test.cc [hunk 37]
```diff
@@ -18,7 +18,7 @@
 #include "starboard/nplb/player_test_fixture.h"
 #include "starboard/nplb/player_test_util.h"
 #include "starboard/nplb/posix_compliance/posix_thread_helpers.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/nplb/player_write_sample_test.cc:21]: remove

File: starboard/nplb/posix_compliance/posix_directory_create_test.cc [hunk 38]
```diff
@@ -15,8 +15,7 @@
 #include <string>
 
 #include "starboard/configuration_constants.h"
-#include "starboard/directory.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/nplb/posix_compliance/posix_directory_create_test.cc:18]: remove

File: starboard/nplb/posix_compliance/posix_directory_get_next_test.cc [hunk 39]
```diff
@@ -21,8 +21,7 @@
 #include "starboard/common/log.h"
 #include "starboard/configuration.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/directory.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/nplb/posix_compliance/posix_directory_get_next_test.cc:24]: remove

File: starboard/nplb/posix_compliance/posix_directory_open_test.cc [hunk 40]
```diff
@@ -17,8 +17,7 @@
 #include <string>
 
 #include "starboard/configuration_constants.h"
-#include "starboard/directory.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/nplb/posix_compliance/posix_directory_open_test.cc:20]: remove

File: starboard/nplb/posix_compliance/posix_file_get_info_test.cc [hunk 41]
```diff
@@ -21,7 +21,7 @@
 #include <string>
 
 #include "starboard/common/time.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/nplb/posix_compliance/posix_file_get_info_test.cc:24]: remove

File: starboard/nplb/posix_compliance/posix_file_get_path_info_test.cc [hunk 42]
```diff
@@ -20,7 +20,7 @@
 
 #include "starboard/common/time.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/nplb/posix_compliance/posix_file_get_path_info_test.cc:23]: remove

File: starboard/nplb/system_get_error_string_test.cc [hunk 43]
```diff
@@ -15,7 +15,7 @@
 #include <fcntl.h>
 
 #include "starboard/common/string.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/nplb/system_get_error_string_test.cc:18]: remove

File: starboard/shared/libde265/de265_video_decoder.cc [hunk 44]
```diff
@@ -17,7 +17,7 @@
 #include "starboard/common/string.h"
 #include "starboard/linux/shared/decode_target_internal.h"
 #include "starboard/shared/libde265/de265_library_loader.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/shared/libde265/de265_video_decoder.cc:20]: remove

File: starboard/shared/libjpeg/jpeg_image_decoder.cc [hunk 45]
```diff
@@ -21,7 +21,7 @@
 
 #include "starboard/common/log.h"
 #include "starboard/configuration.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline starboard/shared/libjpeg/jpeg_image_decoder.cc:24]: remove

File: starboard/shared/media_session/playback_state.cc [hunk 46]
```diff
@@ -18,7 +18,7 @@
 
 #include "starboard/common/log.h"
 #include "starboard/extension/media_session.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/shared/media_session/playback_state.cc:21]: remove

File: starboard/shared/starboard/player/filter/testing/test_util.cc [hunk 47]
```diff
@@ -18,14 +18,13 @@
...
 #include "starboard/common/log.h"
-#include "starboard/directory.h"
 #include "starboard/extension/enhanced_audio.h"
 #include "starboard/shared/starboard/media/media_support_internal.h"
 #include "starboard/shared/starboard/media/mime_type.h"
 #include "starboard/shared/starboard/player/filter/player_components.h"
 #include "starboard/shared/starboard/player/filter/stub_player_components_factory.h"
 #include "starboard/shared/starboard/player/video_dmp_reader.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline starboard/shared/starboard/player/filter/testing/test_util.cc:27]: remove

File: third_party/boringssl/src/config/starboard/openssl/opensslconf.h [hunk 48]
```diff
@@ -4,9 +4,9 @@
 
 #include "starboard/common/log.h"
 #include "starboard/configuration.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline third_party/boringssl/src/config/starboard/openssl/opensslconf.h:7]: remove

File: third_party/boringssl/src/config/starboard/openssl/opensslconf.h [hunk 49]
```diff
@@ -4,9 +4,9 @@
...
 #include "starboard/memory.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline third_party/boringssl/src/config/starboard/openssl/opensslconf.h:9]: remove

File: third_party/crashpad/crashpad/wrapper/wrapper.cc [hunk 50]
```diff
@@ -27,9 +27,8 @@
 #include "client/settings.h"
 #include "starboard/common/system_property.h"
 #include "starboard/configuration_constants.h"
-#include "starboard/directory.h"
 #include "starboard/extension/loader_app_metrics.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline third_party/crashpad/crashpad/wrapper/wrapper.cc:31]: remove

File: third_party/freetype/src/include/freetype/config/ftstdlib.h [hunk 51]
```diff
@@ -114,7 +114,7 @@
 #include <unistd.h>
 
 #include "starboard/common/file_wrapper.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline third_party/freetype/src/include/freetype/config/ftstdlib.h:117]: remove

File: third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h [hunk 52]
```diff
@@ -273,11 +273,9 @@
 #include "starboard/common/log.h"
 #include "starboard/common/spin_lock.h"
 #include "starboard/common/string.h"
-#include "starboard/directory.h"
-#include "starboard/file.h"
+
```
Comment (kaidokert) [inline third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h:276]: remove

File: third_party/libjpeg_turbo/jinclude.h [hunk 53]
```diff
@@ -34,7 +34,7 @@
  */
 
 #ifdef STARBOARD
```
Comment (kaidokert) [inline third_party/libjpeg_turbo/jinclude.h:36]: This ifdef is no longer needed at all, please remove

File: third_party/libxml/starboard/config.h [hunk 54]
```diff
@@ -1,7 +1,7 @@
 #include "starboard/common/log.h"
 #include "starboard/configuration.h"
 #include "starboard/memory.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline third_party/libxml/starboard/config.h:4]: remove

File: third_party/zlib/arm_features.c [hunk 55]
```diff
@@ -21,7 +21,7 @@ int ZLIB_INTERNAL arm_cpu_enable_pmull = 0;
 
 #if defined(STARBOARD)
 #include "starboard/log.h"
-#include "starboard/string.h"
+
```
Comment (kaidokert) [inline third_party/zlib/arm_features.c:24]: remove

**Review by kaidokert — CHANGES_REQUESTED** [review]
Mostly just empty line cleanup nits, and a few empty #ifdef STARBOARDs to get rid of

File: base/files/file_enumerator.h (hunk 1 above)
Comment (madhurajayaraman) [reply base/files/file_enumerator.h:23]: :( I thought the linter would take care of those, will fix!

**Review by y4vor — APPROVED** [review]

**Review by kaidokert — APPROVED** [review]
LGTM

**Pull request merged**